sync/src/block/extension.rs

```diff
@@ -321,7 +321,6 @@ impl ChainNotify for Extension {
         }
         let mut headers_to_download: Vec<_> = enacted
             .into_iter()
-            .chain(new_highest_header.into_iter())
```

Comment:
Would you explain about this change?

Reply:
The purpose of this PR is checking whether the proper proposer creates the proposal or not. Since, If a node downloads a proposal block using block sync, the node could not verify the signature(the block does not contain the signature), I made the block extension not to download the highest block which doesn't be included in the canonical chain(that is proposal block).

Reply:
Is it safe also on other consensus engines?